src/OpenLoco/GameCommands/ChangeCompanyFace.cpp

```diff
@@ -0,0 +1,87 @@
+#include "../CompanyManager.h"
+#include "../Interop/Interop.hpp"
+#include "../Localisation/StringIds.h"
+#include "../Objects/CompetitorObject.h"
+#include "../Objects/ObjectManager.h"
+#include "../Ui/WindowManager.h"
+#include "GameCommands.h"
+
+using namespace OpenLoco::Interop;
+
+namespace OpenLoco::GameCommands
+{
+    // 0x00435506
+    static uint32_t changeCompanyFace(uint8_t flags, CompanyId targetCompanyId, ObjectHeader& targetHeader)
+    {
+        GameCommands::setExpenditureType(ExpenditureType::Miscellaneous);
+
+        // Find target competitor object among loaded objects
+        auto foundCompetitor = ObjectManager::findObjectHandle(targetHeader);
+
+        // See whether any other company is using this competitor already
+        for (auto& company : CompanyManager::companies())
+        {
+            if (company.competitorId == foundCompetitor->id)
+            {
+                if (company.id() != targetCompanyId)
+                {
+                    GameCommands::setErrorText(StringIds::already_selected_for_another_company);
+                    return GameCommands::FAILURE;
+                }
+                else
+                {
+                    // No change; no work required
+                    return 0;
+                }
+            }
+        }
+
+        // Stop here if we're just querying
+        if ((flags & GameCommands::Flags::apply) == 0)
+            return 0;
+
+        // Do we need to load the competitor, still?
+        if (!foundCompetitor)
+        {
+            // Load the new competitor object
```

Comment:
Previously it would first check if there was a free slot and if there wasnt a free slot it would try unload an unallocated to a company slot

Reply:
I noticed it was doing more, but couldn't really see the point. There are only 16 companies, while 32 slots are available. Would the extra check be beneficial in the future?

Reply:
Its not the same check though. The first check is making sure you can't allocate an already in use competitor. The second check is making sure you can't unload a competitor that is in use. In theory you can't get to the second situation due to the first check but I can see why it would be sensible to have both.

Reply:
i put this reply under the wrong thread.